nemo/nemo/core/neural_modules.py

```diff
@@ -115,15 +124,19 @@ def __call__(self, **kwargs):
         #    raise ValueError("We currently do not support calling same NM"
         #                     "more than once")
 
+        # Get input and output ports definitions.
+        input_port_defs = self.input_port_definitions()
+        output_port_defs = self.output_port_definitions()
```

Comment:
why don't use "self.input_port_definitions()" and "self.output_port_definitions()" directly?

Reply:
less changes in code ;)